### ISSUE TITLE
Allow to set specify source files `File Type` using in Xcode to solve C/C++ compile issue

### ISSUE THREAD

**Issue by dreampiggy** [CocoaPods/CocoaPods#8614]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I'm tring to bring a C++ library from CMake to CocoaPods's podspec DSL, to allow using on Apple platforms by CocoaPods. But failed to do so because it does not allow to specify custom file type for the `.cpp` files.

In C/C++/OC language, the `#incldue` can include anything like `*.cpp`, `*.c`. At this time, the file been included can specify whether to be treated as C header/C source files, which have different compile behavior. For example, `.cpp` does not support the `#pragma once` to avoid dunplicated include, only header support.

The original C++ library is here: https://github.com/FLIF-hub/FLIF

## What did you expect to happen?

There should be one syntax in podspec, like this:

```
spec.source_file_types = {
    'flif-interface_common.cpp' => 'sourcecode.cpp.h',
    'flif-interface_enc.cpp' =>  'sourcecode.cpp.h',
    'flif-interface_dec.cpp' =>  'sourcecode.cpp.h',
}
```

And the generated Pod.xcodeproj, should treate these files as the correct file type `C++ Header`

![image](https://user-images.githubusercontent.com/6919743/54485422-10d12a80-48b3-11e9-8dbb-2e1a50cd3b23.png)

## What happened instead?

There are no any syntax, to specify one source files's File Type used by Xcode. So this cause the clang compiler treate the `.cpp` as default source files, which ignore something related to headers like `#pragma once`, so the final result cause a compile error because of the dunplicated include.

### Stack

```
   CocoaPods : 1.6.1
        Ruby : ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin18]
    RubyGems : 2.7.9
        Host : Mac OS X 10.14 (18A391)
       Xcode : 10.1 (10B61)
         Git : git version 2.17.2 (Apple Git-113)
Ruby lib dir : /Users/lizhuoli/.rvm/rubies/ruby-2.5.0/lib
Repositories :
               master - https://github.com/CocoaPods/Specs.git @ 14c522003fccd4ec843d4452075fb9104c096ae2
```

### Installation Source

```
Executable Path: /Users/lizhuoli/.rvm/gems/ruby-2.5.0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.3
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
use_frameworks!

target 'SDWebImageFLIFCoder_Example' do
  platform :ios, '8.0'
  pod 'SDWebImageFLIFCoder', :path => '../'

  target 'SDWebImageFLIFCoder_Tests' do
    inherit! :search_paths

    
  end
end

target 'SDWebImageFLIFCoderExample-macOS' do
  platform :osx, '10.10'
  pod 'SDWebImageFLIFCoder', :path => '../'
end
```

## Project that demonstrates the issue

See the Xcode project which can build successfully:

https://github.com/SDWebImage/libflif-Xcode

The podspec files that can not build successfully because it use the default file type:
https://github.com/SDWebImage/libflif-Xcode/libflif.podspec

**Comment by dnkoutso** [comment]
Not currently supported. Needs a proposal to be approved by other contributors first.

**Comment by dreampiggy** [comment]
Yes...Current I've using a tricky way to solve by the `prepare_command`, which modify the source code to use `.hpp` instead of `.cpp`. It works and I've pushed this repo into the Trunk. But I don't think it's a good idea... 😅 

See [libflif.podspec on CocoaPods](https://github.com/CocoaPods/Specs/blob/master/Specs/7/e/e/libflif/0.3/libflif.podspec.json)

**Comment by Zeaa** [comment]
@dreampiggy  "use .hpp instead of .cpp", solved my problem! 😁
